tools/project_manager/diff.php

```diff
@@ -135,17 +145,41 @@
         $_GET["format"] = "remove";
     }
     $format_url = "?" . attr_safe(http_build_query($_GET));
+    $format_label = html_safe($format_label);
     echo "| <a href='$format_url'>$format_label</a>\n";
+    // Restore $_GET in case needed below
+    $_GET["format"] = $format_save;
+}
+
+if (user_can_mentor_in_any_round()) {
+    // show option to change diff style
+    $bb_diffs_save = array_get($_GET, "bb_diffs", "off");
+    if ($bb_diffs) {
+        $bb_diffs_label = _("Show table-style diff");
+        $_GET["bb_diffs"] = "off";
+    } else {
+        $bb_diffs_label = _("Show BBCode-style diff");
+        $_GET["bb_diffs"] = "on";
+    }
+    $bb_diffs_url = "?" . attr_safe(http_build_query($_GET));
+    $bb_diffs_label = html_safe($bb_diffs_label);
+    echo "| <a href='$bb_diffs_url'>$bb_diffs_label</a>\n";
+    // Restore $_GET in case needed below
+    $_GET["bb_diffs"] = $bb_diffs_save;
 }
 
 echo "</p>\n";
 
 // ---------------------------------------------------------
 
-$diffEngine = new DifferenceEngineWrapper();
+$diffurl = "$code_url/tools/project_manager/diff.php?project=$projectid&image=$image&L_round_num=$L_round_num&R_round_num=$R_round_num";
+
+$diffEngine = $bb_diffs ? new DifferenceEngineWrapperBBHTML($diffurl) : new DifferenceEngineWrapperTable();
 
 $diffEngine->setText($L_text, $R_text);
-$diffEngine->showDiff($L_label, $R_label);
+$difftext = $diffEngine->getDiff($L_label, $R_label);
```

Comment:
You might consider just echoing here instead of storing it in a variable and then writing it out.

Reply:
Thanks - should be OK now